pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -279,8 +279,15 @@ func (self *RefsHelper) NewBranch(from string, fromFormattedName string, suggest
 		InitialContent: suggestedBranchName,
 		HandleConfirm: func(response string) error {
 			self.c.LogAction(self.c.Tr.Actions.CreateBranch)
-			if err := self.c.Git().Branch.New(SanitizedBranchName(response), from); err != nil {
-				return err
+			newBranchName := SanitizedBranchName(response)
+			if newBranchName == suggestedBranchName {
+				if err := self.c.Git().Branch.New(newBranchName, from); err != nil {
+					return err
+				}
+			} else {
+				if err := self.c.Git().Branch.NewWithoutTracking(newBranchName, from); err != nil {
+					return err
+				}
```

Comment:
The duplicated `if` statement bothers me slightly; how do you feel about doing this instead?
```
	newBranchFunc := self.c.Git().Branch.New
	if newBranchName != suggestedBranchName {
		newBranchFunc = self.c.Git().Branch.NewWithoutTracking
	}
	if err := newBranchFunc(newBranchName, from); err != nil {
		return err
	}
```
Maybe that's too obscure, I'm not sure.

Another option would be to add an argument to the `New` function; it could be called something like `inhibitAutoTracking bool`. I'm a little on the fence here, but I think I slightly prefer that. It has the downside that you need to touch all other call sites, but on the other hand that may be seen as a pro argument: it forces you to consider what's the right thing to do at all other call sites.

Just input, let me know what you think.

Reply:
> The duplicated if statement bothers me slightly; how do you feel about doing this instead?
> ``` go
> newBranchFunc := self.c.Git().Branch.New
> if newBranchName != suggestedBranchName {
> 	newBranchFunc = self.c.Git().Branch.NewWithoutTracking
> }
> if err := newBranchFunc(newBranchName, from); err != nil {
> 	return err
> }
> ```

I am fine with that, I don't think it's too obscure.